server/src/test/java/org/opensearch/common/settings/SettingsModuleTests.java

```diff
@@ -237,4 +237,28 @@ public void testOldMaxClauseCountSetting() {
             ex.getMessage()
         );
     }
+
+    public void testDynamicNodeSettingsRegistration() {
+        Settings settings = Settings.builder().put("some.custom.setting", "2.0").build();
+        SettingsModule module = new SettingsModule(settings, Setting.floatSetting("some.custom.setting", 1.0f, Property.NodeScope));
+        assertNotNull(module.getClusterSettings().get("some.custom.setting"));
+        // For unregistered setting the value is expected to be null
+        assertNull(module.getClusterSettings().get("some.custom.setting2"));
+        assertInstanceBinding(module, Settings.class, (s) -> s == settings);
+
+        module.registerDynamicSetting(Setting.floatSetting("some.custom.setting2", 1.0f, Property.NodeScope));
+        assertNotNull(module.getClusterSettings().get("some.custom.setting2"));
```

Comment:
Any way to add additional tests that there are now two settings here? At least check that adding the second setting didn't remove the first. (Same comment line 262.)